internal/mode/static/state/graph/backend_refs.go

```diff
@@ -142,7 +148,7 @@ func createBackendRef(
 		return backendRef, &cond
 	}
 
-	svcNsName, svcPort, err := getServiceAndPortFromRef(ref.BackendRef, sourceNamespace, services, refPath)
+	svcNsName, svcPort, svcIPFamilies, err := getServiceAndPortFromRef(ref.BackendRef, sourceNamespace, services, refPath)
```

Comment:
I think it would be better to just pass `svc.Spec.IPFamilies, servicePort, err` and we have all the information to get service name in the function `createBackendRef` instead of sending the entire `svc` info